mongodump/oplog_dump.go

```diff
@@ -86,16 +85,13 @@ func (dump *MongoDump) checkOplogTimestampExists(ts bson.MongoTimestamp) (bool,
 }
 
 func oplogDocumentFilter(in []byte) ([]byte, error) {
-	var rawD bson.RawD
-	err := bson.Unmarshal(in, &rawD)
-	if err != nil {
-		return nil, err
-	}
-
 	var nsD struct {
 		NS string `bson:"ns"`
+		O *struct {
+			RenameCollection interface{} `bson:"renameCollection"`
+		} `bson:"o"`
 	}
-	err = bson.Unmarshal(in, &nsD)
+	err := bson.Unmarshal(in, &nsD)
```

Comment:
I find this approach of defining a struct that doesn't apply to all oplog entries and unmarshalling to see what worked to be a bit odd/confusing.  I think a better approach in this functions is to convert the `in` parameter to a `bson.Raw` and use `Lookup`/`LookupErr` to spot-check the conditions we care about without unmarshaling to struct.  (It's possibly higher performance, too.)